tc/core/utils/time.h

```diff
@@ -18,7 +18,61 @@
 #include <chrono>
 
 namespace tc {
-using Duration = std::chrono::high_resolution_clock::duration;
+struct Duration {
+ private:
+  explicit Duration(size_t us) : val_(us) {}
+  explicit Duration(std::chrono::microseconds us) : val_(us) {}
+
+ public:
+  Duration() : val_(std::chrono::microseconds::max()) {}
+  Duration(const Duration& d) : val_(d.val_) {}
+
+  static inline Duration fromStartStop(
+      std::chrono::time_point<std::chrono::system_clock> begin,
+      std::chrono::time_point<std::chrono::system_clock> end) {
```

Comment:
The second argument to `fromStartStop` is always `std::chrono::system_clock::now()`.
Drop the second argument and rename (say, to `since`).